crates/hyperdrive-math/src/long/close.rs

```diff
@@ -1,32 +1,73 @@
 use fixed_point::FixedPoint;
 use fixed_point_macros::fixed;
 
-use crate::{State, YieldSpace};
+use crate::{
+    close_long_curve_fee, close_long_flat_fee,
+    yield_space::calculate_shares_out_given_bonds_in_down, State, YieldSpace,
+};
+
+fn calculate_close_long_flat_plus_curve<F: Into<FixedPoint>>(
+    ze: F,
+    y: F,
+    c: F,
+    mu: F,
+    t: F,
+    dy: F,
+    tr: F,
+) -> FixedPoint {
+    let [ze, y, c, mu, t, dy, tr] = [ze, y, c, mu, t, dy, tr].map(|v| v.into());
```

Comment:
One thing to watch out for with this kind of trait declaration is that `F` is bound to _one_ kind of `Into<FixedPoint>`. So if you have `a: I256` and `b: U256`, you won't be able to call `calculate_close_long_flat_plus_curve(a, b)` since they aren't the same `F`. It's gross, but if we want to preserve this functionality, we'd need `F1`, `F2`, etc. 

Here's an example so that you can see what I'm talking about: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a97ab399342a14ed0e66d748b5be57d9.

Reply:
It's to make it easier to call the function. If you have a `U256` and the argument is a `FixedPoint`, you have to know that and cast it yourself. This pattern allows you to just pass the parameter in directly without worrying about conversions.

This pattern was initially used in all of the publicly accessible functions. We've made more and more of the Rust SDK public, and I would agree that we've dropped some consistency. We should recover that. FWIW, I think a lot of the functions that are public are poorly named/documented, which is something that we'll want to keep an eye on.